internal/zinc-scripted/src/test/scala/sbt/internal/inc/IncHandler.scala

```diff
@@ -332,7 +349,13 @@ case class ProjectStructure(
     def relativeClassDir(f: File): File = f.relativeTo(classesDir) getOrElse f
     def products(srcFile: String): Set[String] = {
       val productFiles = analysis.relations.products(baseDirectory / srcFile)
-      productFiles.map(relativeClassDir).map(_.getPath)
+      productFiles.map { file =>
+        if (JarUtils.isClassInJar(file)) {
+          JarUtils.ClassInJar.fromFile(file).relClass
+        } else {
+          relativeClassDir(file).getPath.replace('\\', '/')
```

Comment:
I imagine this is for Windows? Can we make it clear by guarding against it if it's not running in a Windows machine?